subprojects/docs/src/docs/userguide/commandLineInterface.adoc

```diff
@@ -0,0 +1,520 @@
+    <sourcefile file="subdir/myproject.gradle"/>
+    <output args="-q -b subdir/myproject.gradle hello"/>
+</sample>
+++++
+
+Alternatively, you can use the `-p` option to specify the project directory to use. For multi-project builds you should use `-p` option instead of `-b` option.
+
+++++
+<sample id="selectProjectUsingProjectDir" dir="userguide/tutorial/selectProject" title="Selecting the project using project directory">
+    <output args="-q -p subdir hello"/>
+</sample>
+++++
+
+[[sec:command_line_completion]]
+=== Command-Line Completion
+
+Gradle provides bash and zsh tab completion support for tasks, options, and Gradle properties through https://github.com/gradle/gradle-completion[gradle-completion], installed separately.
+
+++++
+<figure>
+    <title>Gradle Completion</title>
+    <imageobject>
+        <imagedata fileref="img/gradle-completion-4.0.gif"/>
+    </imageobject>
+</figure>
+++++
+
+[[sec:command_line_logging]]
+=== Logging Options
+
+==== Setting Log Level
+You can customize the verbosity of Gradle logging with the following options, ordered from least verbose to most verbose. Learn more in the <<logging, logging documentation>>.
+
+`-Dorg.gradle.logging.level=(quiet warn lifecycle info debug)`::
+Set logging level via Gradle properties.
+
+`-q`, `--quiet`::
+Log errors only.
+
+`-w`, `--warn`::
+Set log level to warn.
+
+`-i`, `--info`::
+Set log level to info.
+
+`-d`, `--debug`::
+Log in debug mode (includes normal stacktrace).
+
+Lifecycle is the default log level.
+
+==== Customizing Log Format
+You can control the use of rich output (colors and font variants) by specifying the "console" mode in the following ways:
+
+`-Dorg.gradle.console=(auto plain rich verbose)`::
+Specify console mode via Gradle properties. Different modes described immediately below.
+
+`--console=(auto plain rich verbose)`::
+Specifies which type of console output to generate.
++
+Set to `plain` to generate plain text only. This option disables all color and other rich output in the console output.
++
+Set to `auto` (the default) to enable color and other rich output in the console output when the build process is attached to a console, or to generate plain text only when not attached to a console.
++
+Set to `rich` to enable color and other rich output in the console output, regardless of whether the build process is not attached to a console. When not attached to a console, the build output will use ANSI control characters to generate the rich output.
++
+Set to `verbose` to enable color and other rich output like the `rich`, but output task names and outcomes at the lifecycle log level, as is done by default in Gradle 3.5 and earlier.
+
+==== Rich Console
+
+Gradle's rich console displays extra information while builds are running.
+
+image::img/console-animation.gif[alt="Gradle Rich Console",data-alt="foo.gif",class="js-gif-playback-control gif-control gif-control__paused"]
+
+Features:
```

Comment:
:x: I am not convinced that readers will know what all of these terms mean e.g. work-in-progress lines.

Reply:
Agreed. I'll annotate this somehow instead.

Reply:
Nice, I like it.

Reply:
Thanks :)